src/coreclr/gc/gc.cpp

```diff
@@ -20839,6 +20839,11 @@ int gc_heap::joined_generation_to_condemn (BOOL should_evaluate_elevation,
         }
     }
 
+    if (settings.reason == reason_induced_aggressive)
```

Comment:
this should do `gc_data_global.gen_to_condemn_reasons.set_condition` like with the other reasons.

also now I'm looking at the code, it doesn't seem to set the condemn generation number anywhere for this mode? what is the behavior if you do do `GC.Collect (0, GCCollectionMode.Aggressive)?`? 0 is replaced by 2? if so I don't see this in the documentation. it should that this mode just does a full compacting GC.